### ISSUE TITLE
running hasura migrations using the cli docker container does not return a non-zero exit if a migration fails

### ISSUE THREAD

**Issue by khumps** [hasura/graphql-engine#7698]
### Version Information

CLI Version (for CLI related issue): hasura/graphql-engine:v2.0.9.cli-migrations-v3

### Environment

AWS linux v2

### What is the expected behaviour?

up.sql:
```
INVALID SQL HERE
```
running `docker container run -it --tty  -e HASURA_GRAPHQL_ENDPOINT="http://hasura:8080" -e HASURA_GRAPHQL_ADMIN_SECRET="$GRAPHQL_SECRET" -e HASURA_GRAPHQL_MIGRATIONS_DIRECTORY="./migrations" -e HASURA_GRAPHQL_METADATA_DIRECTORY="./metadata" -v ${CURRDIR}services/hasura/migration:/migration --entrypoint /bin/hasura-cli "hasura/graphql-engine:v2.0.9.cli-migrations-v3" migrate apply --project /migration --database-name default` outputs 

```
INFO Help us improve Hasura! The cli collects anonymized usage stats which
allow us to keep improving Hasura at warp speed. To opt-out or read more,
visit https://hasura.io/docs/latest/graphql/core/guides/telemetry.html 
ERRO skipping applying migrations on database default, encountered:                                                                                                                                                     
{
  "internal": {
    "statement": "SELCT * FROM layer0_data.cable;",
    "prepared": false,
    "error": {
      "exec_status": "FatalError",
      "hint": null,
      "message": "syntax error at or near \"SELCT\"",
      "status_code": "42601",
      "description": null
    },
    "arguments": []
  },
  "path": "$",
  "error": "query execution failed",
  "code": "postgres-error"
} 
➜  echo $?
0
```

The migration failed but the exit code of the docker container is 0

I tried running the same test using the local hasura cli on my mac and it returned a non-zero exit as expected

**Comment by waterdrop01** [comment]
I think you hit the same issue than me: https://github.com/hasura/graphql-engine/issues/7629

Which is already being tracked (and fixed) here: https://github.com/hasura/graphql-engine/issues/7499

**Comment by khumps** [comment]
> I think you hit the same issue than me: #7629
> 
> Which is already being tracked (and fixed) here: #7499

I actually saw your issues before I posted. If I was reading them right, the fix they proposed for yours doesn't appear to affect the logic for the `--database-name` flag

**Comment by khumps** [comment]
I have tested this with v2.1.0 of the container and am no longer able to reproduce
